jina/types/arrays/bpm.py

```diff
@@ -0,0 +1,101 @@
+from collections import OrderedDict
+from typing import Union
+
+from ... import Document
+
+if False:
+    from .memmap import DocumentArrayMemmap
+
+
+class BufferPoolManager:
```

Comment:
this class requires specific isolated `unittests`

Reply:
I added unit tests in test_buffer.py but the problem is that the class is coupled with DAM to access disk. In fact the code would be cleaner if I moved all disk operations to another class (DiskManager), and there would be less coupling: 
DAM <== BufferPoolManager <== DiskManager
what do you suggest here?

Reply:
but anyway u would be coupled with another, I think it would be overkilling. It can be unittested usning `DocumentArrayMemmap` as input, that's clear

Reply:
I added separate unit tests to `BufferPoolManager`